### ISSUE TITLE
TYPEDEF STRUCT and FIELDALIGN:16

### ISSUE THREAD

**Issue by vid512** [Terraspace/UASM#139]
There seems to be some problem with aligning a structure member, which is a typedef of structure, to 16, as if the inner structure was 16-byte type.

```
.code

OPTION FIELDALIGN:16

INNER1 struct
a1 QWORD ?
a2 QWORD ?
INNER1 ends

INNER2 typedef INNER1

S1 struct
foo QWORD ?
inn INNER1  <>      ;member offset = 0x8
S1 ends

S2 struct
foo QWORD ?
inn INNER2 <>       ;member offset = 0x10
S2 ends

end
```

listing confirms this:

```
00000000                        S1 struct
00000000                         foo QWORD ?
00000008                         inn INNER1  <>      ;member offset = 0x8
00000018                        S1 ends

00000000                        S2 struct
00000000                         foo QWORD ?
00000010                         inn INNER2 <>       ;member offset = 0x10
00000020                        S2 ends
```

**Comment by john-terraspace** [comment]
Ok, so look at this with OPTION FIELDALIGN:16

INNER1 should have
INNER1.a1   ; offset +0
INNER1.a2   ; offset +16

Then we have the typedef

Then both S1 and S2 should produce the same result which is:

Sx.foo      ; offset +0
Sx.inn.a1  ; offset +16
Sx.inn.a2  ; offset +32

Does that sound right?

**Comment by vid512** [comment]
Seems to me, there is some confusion about what the "OPTION FIELDALIGN" actually does. According to documentation (), it is the default align for member values. 

However in practice, until now it worked differently. It was the maximum alignment allowed for member in order to reach its natural alignment. Best showcased by following listing:
```

                                OPTION FIELDALIGN:4

00000000                        S1 struct
00000000                         m1 DWORD ?             ;+0x00
00000004                         m2 QWORD ?             ;+0x04 = unaligned
0000000C                         m3 QWORD ?             ;+0x0C = unaligned
00000014                         m4 XMMWORD ?           ;+0x14 = unaligned
00000024                        S1 ends

                                OPTION FIELDALIGN:8

00000000                        S2 struct
00000000                         m1 DWORD ?             ;+0x00
00000008                         m2 QWORD ?             ;+0x08 = aligned
00000010                         m3 QWORD ?             ;+0x10 = aligned
00000018                         m4 XMMWORD ?           ;+0x18 = unaligned
00000028                        S2 ends

                                OPTION FIELDALIGN:16

00000000                        S3 struct
00000000                         m1 DWORD ?             ;+0x00
00000008                         m2 QWORD ?             ;+0x08 = aligned
00000010                         m3 QWORD ?             ;+0x10 = aligned
00000020                         m4 XMMWORD ?           ;+0x20 = aligned
00000030                        S3 ends
```

What I wanted is to have each member naturally aligned (to its own size). Hence I used "OPTION FIELDALIGN:16", based on empirical experiments. Up until the reported problem with TYPEDEF STRUCT, it did what I wanted.

If OPTION FIELDALIGN is supposed to do what you wrote, and current behavior was a bug, then please include some way to force natural alignment for each structure member.

**Comment by john-terraspace** [comment]
I think I agree with your interpretation more.

Any value should be aligned to it's natural size relative to the base of the struct, the fieldalign name is silly really in the context as it really means maximum alignment size in the structure. Maybe we should add new options for this to make it more clear and make fieldalign do whatever masm does.

I'd almost be inclined to say they're parameters added to the struct definition:

S1 STRUCT:{1-32} .. 
0 would be align each field to it's natural size
1 would say each field is byte aligned, or as declared
2-32 would force that specific alignment to each and every field

Something like that?

**Comment by vid512** [comment]
I think that 99.9% of time, you don't need to change member alignment type per-structure. Usually you want same format for all structures, or for a very big set of structures. For that, OPTION / PUSHCONTEXT / POPCONTEXT is ideal.

I'd be very afraid to change FIELDALIGN now, because of breaking backwards compatibility with existing code.

I think this is a case of bad naming and bad documentation, not case of bad functionality. The actual intended behavior is IMO described here (note the same /Zp option): https://docs.microsoft.com/en-us/cpp/preprocessor/pack?view=vs-2019

My suggestion is to solve it this way: 
- fix the description in docs
- keed FIELDALIGN as is
- fix the reported problem of TYPEDEF STRUCT member behaving differently than STRUCT member
- if you really really feel the need to clear up confusion about name of option for future, then declare it as obsolete (with warning not error), and add a new PACK/FIELDPACK/MEMBERPACK/STRUCTPACK/etc. option with same functionality. That will make it clear this is the same functionality as C pragma pack.

**Comment by john-terraspace** [comment]
Fixed in 2.55

`UASM v2.55, Mar 19 2022, Masm-compatible assembler.

struct_typedef_align.asm
                                .x64
                                .model flat

                                .code

                                OPTION FIELDALIGN:16

00000000                        INNER1 struct
00000000                        a1 QWORD ?
00000008                        a2 QWORD ?
00000010                        INNER1 ends

                                INNER2 typedef INNER1

00000000                        S1 struct
00000000                        foo QWORD ?
00000008                        inn INNER1  <>      ;member offset = 0x8
00000018                        S1 ends

00000000                        S2 struct
00000000                        foo QWORD ?
00000008                        inn INNER2 <>       ;member offset = 0x10
00000018                        S2 ends

                                end


Binary Map:

Segment                  Pos(file)     RVA  Size(fil) Size(mem)
---------------------------------------------------------------
_TEXT                           0        0         0         0
_DATA                           0        0         0         0
---------------------------------------------------------------
                                                   0         0


Macros:

                N a m e                 Type

@CatStr  . . . . . . . . . . . .        Func     0
@Environ . . . . . . . . . . . .        Func     0
@InStr . . . . . . . . . . . . .        Func     0
@SizeStr . . . . . . . . . . . .        Func     0
@SubStr  . . . . . . . . . . . .        Func     0
ALIGNADDR  . . . . . . . . . . .        Proc     0
ASDOUBLE . . . . . . . . . . . .        Func     0
ASFLOAT  . . . . . . . . . . . .        Func     0
CLASS  . . . . . . . . . . . . .        Proc     0
CMETHOD  . . . . . . . . . . . .        Proc     0
COMINTERFACE . . . . . . . . . .        Proc     0
CSTATIC  . . . . . . . . . . . .        Proc     0
CSTR . . . . . . . . . . . . . .        Func     0
CVIRTUAL . . . . . . . . . . . .        Proc     0
ENDCLASS . . . . . . . . . . . .        Proc     0
ENDCOMINTERFACE  . . . . . . . .        Proc     0
ENDMETHOD  . . . . . . . . . . .        Proc     0
ENDMETHODS . . . . . . . . . . .        Proc     0
ENDOINTERFACE  . . . . . . . . .        Proc     0
ENDRAWINTERFACE  . . . . . . . .        Proc     0
EXPAND_PREFIX  . . . . . . . . .        Func     0
FP10 . . . . . . . . . . . . . .        Func     0
FP4  . . . . . . . . . . . . . .        Func     0
FP8  . . . . . . . . . . . . . .        Func     0
GETMASK128 . . . . . . . . . . .        Proc     0
LOADMSD  . . . . . . . . . . . .        Proc     0
LOADMSS  . . . . . . . . . . . .        Proc     0
LOADPS . . . . . . . . . . . . .        Proc     0
LOADSD . . . . . . . . . . . . .        Proc     0
LOADSS . . . . . . . . . . . . .        Proc     0
MEMALLOC . . . . . . . . . . . .        Proc     0
MEMFREE  . . . . . . . . . . . .        Proc     0
METHOD . . . . . . . . . . . . .        Proc     0
MOV128 . . . . . . . . . . . . .        Proc     0
MOV64  . . . . . . . . . . . . .        Proc     0
MOVXMMR128 . . . . . . . . . . .        Proc     0
NOTMASK128 . . . . . . . . . . .        Proc     0
OINTERFACE . . . . . . . . . . .        Proc     0
R4P  . . . . . . . . . . . . . .        Func     0
R8P  . . . . . . . . . . . . . .        Func     0
RAWINTERFACE . . . . . . . . . .        Proc     0
REGS15STORAGE  . . . . . . . . .        Proc     0
REPARG . . . . . . . . . . . . .        Func     0
RV . . . . . . . . . . . . . . .        Func     0
SHIFTLEFT128 . . . . . . . . . .        Proc     0
SHIFTRIGHT128  . . . . . . . . .        Proc     0
SLXMMR . . . . . . . . . . . . .        Proc     0
SRXMMR . . . . . . . . . . . . .        Proc     0
STATICMETHOD . . . . . . . . . .        Proc     0
STATICVECMETHOD  . . . . . . . .        Proc     0
STDFUNC  . . . . . . . . . . . .        Proc     0
UINVOKE  . . . . . . . . . . . .        Func     0
VECMETHOD  . . . . . . . . . . .        Proc     0
WSTR . . . . . . . . . . . . . .        Func     0
_ARRAY . . . . . . . . . . . . .        Func     0
_DECLARE . . . . . . . . . . . .        Proc     0
_DELETE  . . . . . . . . . . . .        Proc     0
_DELETEARRAY . . . . . . . . . .        Proc     0
_DEREF . . . . . . . . . . . . .        Proc     0
_DEREFI  . . . . . . . . . . . .        Func     0
_DEREFR  . . . . . . . . . . . .        Proc     0
_DEREFRR . . . . . . . . . . . .        Proc     0
_I . . . . . . . . . . . . . . .        Func     0
_INVOKE  . . . . . . . . . . . .        Proc     0
_ITEM  . . . . . . . . . . . . .        Func     0
_ITEMR . . . . . . . . . . . . .        Func     0
_NEW . . . . . . . . . . . . . .        Func     0
_RBXNEW  . . . . . . . . . . . .        Func     0
_SINVOKE . . . . . . . . . . . .        Proc     0
_STATIC  . . . . . . . . . . . .        Func     0
_STATICREF . . . . . . . . . . .        Proc     0
_V . . . . . . . . . . . . . . .        Func     0
_VINVOKE . . . . . . . . . . . .        Proc     0
arginvoke  . . . . . . . . . . .        Func     0


Structures and Unions:

                N a m e                 Size/Ofs   Type

INNER1 . . . . . . . . . . . . .              10 (16)
  a1 . . . . . . . . . . . . . .               0   QWord
  a2 . . . . . . . . . . . . . .               8   QWord
S1 . . . . . . . . . . . . . . .              18 (16)
  foo  . . . . . . . . . . . . .               0   QWord
  inn  . . . . . . . . . . . . .               8   INNER1
S2 . . . . . . . . . . . . . . .              18 (16)
  foo  . . . . . . . . . . . . .               0   QWord
  inn  . . . . . . . . . . . . .               8   INNER2
__m128 . . . . . . . . . . . . .              10
  f32  . . . . . . . . . . . . .               0   __m128f
  i8 . . . . . . . . . . . . . .               0   __m128b
  i16  . . . . . . . . . . . . .               0   __m128w
  i32  . . . . . . . . . . . . .               0   __m128i
  d64  . . . . . . . . . . . . .               0   __m128d
  q64  . . . . . . . . . . . . .               0   __m128q
__m128b  . . . . . . . . . . . .              10
  b0 . . . . . . . . . . . . . .               0   Byte
  b1 . . . . . . . . . . . . . .               1   Byte
  b2 . . . . . . . . . . . . . .               2   Byte
  b3 . . . . . . . . . . . . . .               3   Byte
  b4 . . . . . . . . . . . . . .               4   Byte
  b5 . . . . . . . . . . . . . .               5   Byte
  b6 . . . . . . . . . . . . . .               6   Byte
  b7 . . . . . . . . . . . . . .               7   Byte
  b8 . . . . . . . . . . . . . .               8   Byte
  b9 . . . . . . . . . . . . . .               9   Byte
  b10  . . . . . . . . . . . . .               A   Byte
  b11  . . . . . . . . . . . . .               B   Byte
  b12  . . . . . . . . . . . . .               C   Byte
  b13  . . . . . . . . . . . . .               D   Byte
  b14  . . . . . . . . . . . . .               E   Byte
  b15  . . . . . . . . . . . . .               F   Byte
__m128d  . . . . . . . . . . . .              10
  d0 . . . . . . . . . . . . . .               0   
  d1 . . . . . . . . . . . . . .               8   
__m128f  . . . . . . . . . . . .              10
  f0 . . . . . . . . . . . . . .               0   
  f1 . . . . . . . . . . . . . .               4   
  f2 . . . . . . . . . . . . . .               8   
  f3 . . . . . . . . . . . . . .               C   
__m128i  . . . . . . . . . . . .              10
  i0 . . . . . . . . . . . . . .               0   DWord
  i1 . . . . . . . . . . . . . .               4   DWord
  i2 . . . . . . . . . . . . . .               8   DWord
  i3 . . . . . . . . . . . . . .               C   DWord
__m128q  . . . . . . . . . . . .              10
  q0 . . . . . . . . . . . . . .               0   QWord
  q1 . . . . . . . . . . . . . .               8   QWord
__m128w  . . . . . . . . . . . .              10
  w0 . . . . . . . . . . . . . .               0   Word
  w1 . . . . . . . . . . . . . .               2   Word
  w2 . . . . . . . . . . . . . .               4   Word
  w3 . . . . . . . . . . . . . .               6   Word
  w4 . . . . . . . . . . . . . .               8   Word
  w5 . . . . . . . . . . . . . .               A   Word
  w6 . . . . . . . . . . . . . .               C   Word
  w7 . . . . . . . . . . . . . .               E   Word
__m256 . . . . . . . . . . . . .              20
  f32  . . . . . . . . . . . . .               0   __m256f
  i8 . . . . . . . . . . . . . .               0   __m256b
  i16  . . . . . . . . . . . . .               0   __m256w
  i32  . . . . . . . . . . . . .               0   __m256i
  d64  . . . . . . . . . . . . .               0   __m256d
  q64  . . . . . . . . . . . . .               0   __m256q
__m256b  . . . . . . . . . . . .              20
  b0 . . . . . . . . . . . . . .               0   Byte
  b1 . . . . . . . . . . . . . .               1   Byte
  b2 . . . . . . . . . . . . . .               2   Byte
  b3 . . . . . . . . . . . . . .               3   Byte
  b4 . . . . . . . . . . . . . .               4   Byte
  b5 . . . . . . . . . . . . . .               5   Byte
  b6 . . . . . . . . . . . . . .               6   Byte
  b7 . . . . . . . . . . . . . .               7   Byte
  b8 . . . . . . . . . . . . . .               8   Byte
  b9 . . . . . . . . . . . . . .               9   Byte
  b10  . . . . . . . . . . . . .               A   Byte
  b11  . . . . . . . . . . . . .               B   Byte
  b12  . . . . . . . . . . . . .               C   Byte
  b13  . . . . . . . . . . . . .               D   Byte
  b14  . . . . . . . . . . . . .               E   Byte
  b15  . . . . . . . . . . . . .               F   Byte
  b16  . . . . . . . . . . . . .              10   Byte
  b17  . . . . . . . . . . . . .              11   Byte
  b18  . . . . . . . . . . . . .              12   Byte
  b19  . . . . . . . . . . . . .              13   Byte
  b20  . . . . . . . . . . . . .              14   Byte
  b21  . . . . . . . . . . . . .              15   Byte
  b22  . . . . . . . . . . . . .              16   Byte
  b23  . . . . . . . . . . . . .              17   Byte
  b24  . . . . . . . . . . . . .              18   Byte
  b25  . . . . . . . . . . . . .              19   Byte
  b26  . . . . . . . . . . . . .              1A   Byte
  b27  . . . . . . . . . . . . .              1B   Byte
  b28  . . . . . . . . . . . . .              1C   Byte
  b29  . . . . . . . . . . . . .              1D   Byte
  b30  . . . . . . . . . . . . .              1E   Byte
  b31  . . . . . . . . . . . . .              1F   Byte
__m256d  . . . . . . . . . . . .              20
  d0 . . . . . . . . . . . . . .               0   
  d1 . . . . . . . . . . . . . .               8   
  d2 . . . . . . . . . . . . . .              10   
  d3 . . . . . . . . . . . . . .              18   
__m256f  . . . . . . . . . . . .              20
  f0 . . . . . . . . . . . . . .               0   
  f1 . . . . . . . . . . . . . .               4   
  f2 . . . . . . . . . . . . . .               8   
  f3 . . . . . . . . . . . . . .               C   
  f4 . . . . . . . . . . . . . .              10   
  f5 . . . . . . . . . . . . . .              14   
  f6 . . . . . . . . . . . . . .              18   
  f7 . . . . . . . . . . . . . .              1C   
__m256i  . . . . . . . . . . . .              20
  i0 . . . . . . . . . . . . . .               0   DWord
  i1 . . . . . . . . . . . . . .               4   DWord
  i2 . . . . . . . . . . . . . .               8   DWord
  i3 . . . . . . . . . . . . . .               C   DWord
  i4 . . . . . . . . . . . . . .              10   DWord
  i5 . . . . . . . . . . . . . .              14   DWord
  i6 . . . . . . . . . . . . . .              18   DWord
  i7 . . . . . . . . . . . . . .              1C   DWord
__m256q  . . . . . . . . . . . .              20
  q0 . . . . . . . . . . . . . .               0   QWord
  q1 . . . . . . . . . . . . . .               8   QWord
  q2 . . . . . . . . . . . . . .              10   QWord
  q3 . . . . . . . . . . . . . .              18   QWord
__m256w  . . . . . . . . . . . .              20
  w0 . . . . . . . . . . . . . .               0   Word
  w1 . . . . . . . . . . . . . .               2   Word
  w2 . . . . . . . . . . . . . .               4   Word
  w3 . . . . . . . . . . . . . .               6   Word
  w4 . . . . . . . . . . . . . .               8   Word
  w5 . . . . . . . . . . . . . .               A   Word
  w6 . . . . . . . . . . . . . .               C   Word
  w7 . . . . . . . . . . . . . .               E   Word
  w8 . . . . . . . . . . . . . .              10   Word
  w9 . . . . . . . . . . . . . .              12   Word
  w10  . . . . . . . . . . . . .              14   Word
  w11  . . . . . . . . . . . . .              16   Word
  w12  . . . . . . . . . . . . .              18   Word
  w13  . . . . . . . . . . . . .              1A   Word
  w14  . . . . . . . . . . . . .              1C   Word
  w15  . . . . . . . . . . . . .              1E   Word
__m512 . . . . . . . . . . . . .              40
  f32  . . . . . . . . . . . . .               0   __m512f
  i8 . . . . . . . . . . . . . .               0   __m512b
  i16  . . . . . . . . . . . . .               0   __m512w
  i32  . . . . . . . . . . . . .               0   __m512i
  d64  . . . . . . . . . . . . .               0   __m512d
  q64  . . . . . . . . . . . . .               0   __m512q
__m512b  . . . . . . . . . . . .              40
  b0 . . . . . . . . . . . . . .               0   Byte
  b1 . . . . . . . . . . . . . .               1   Byte
  b2 . . . . . . . . . . . . . .               2   Byte
  b3 . . . . . . . . . . . . . .               3   Byte
  b4 . . . . . . . . . . . . . .               4   Byte
  b5 . . . . . . . . . . . . . .               5   Byte
  b6 . . . . . . . . . . . . . .               6   Byte
  b7 . . . . . . . . . . . . . .               7   Byte
  b8 . . . . . . . . . . . . . .               8   Byte
  b9 . . . . . . . . . . . . . .               9   Byte
  b10  . . . . . . . . . . . . .               A   Byte
  b11  . . . . . . . . . . . . .               B   Byte
  b12  . . . . . . . . . . . . .               C   Byte
  b13  . . . . . . . . . . . . .               D   Byte
  b14  . . . . . . . . . . . . .               E   Byte
  b15  . . . . . . . . . . . . .               F   Byte
  b16  . . . . . . . . . . . . .              10   Byte
  b17  . . . . . . . . . . . . .              11   Byte
  b18  . . . . . . . . . . . . .              12   Byte
  b19  . . . . . . . . . . . . .              13   Byte
  b20  . . . . . . . . . . . . .              14   Byte
  b21  . . . . . . . . . . . . .              15   Byte
  b22  . . . . . . . . . . . . .              16   Byte
  b23  . . . . . . . . . . . . .              17   Byte
  b24  . . . . . . . . . . . . .              18   Byte
  b25  . . . . . . . . . . . . .              19   Byte
  b26  . . . . . . . . . . . . .              1A   Byte
  b27  . . . . . . . . . . . . .              1B   Byte
  b28  . . . . . . . . . . . . .              1C   Byte
  b29  . . . . . . . . . . . . .              1D   Byte
  b30  . . . . . . . . . . . . .              1E   Byte
  b31  . . . . . . . . . . . . .              1F   Byte
  b32  . . . . . . . . . . . . .              20   Byte
  b33  . . . . . . . . . . . . .              21   Byte
  b34  . . . . . . . . . . . . .              22   Byte
  b35  . . . . . . . . . . . . .              23   Byte
  b36  . . . . . . . . . . . . .              24   Byte
  b37  . . . . . . . . . . . . .              25   Byte
  b38  . . . . . . . . . . . . .              26   Byte
  b39  . . . . . . . . . . . . .              27   Byte
  b40  . . . . . . . . . . . . .              28   Byte
  b41  . . . . . . . . . . . . .              29   Byte
  b42  . . . . . . . . . . . . .              2A   Byte
  b43  . . . . . . . . . . . . .              2B   Byte
  b44  . . . . . . . . . . . . .              2C   Byte
  b45  . . . . . . . . . . . . .              2D   Byte
  b46  . . . . . . . . . . . . .              2E   Byte
  b47  . . . . . . . . . . . . .              2F   Byte
  b48  . . . . . . . . . . . . .              30   Byte
  b49  . . . . . . . . . . . . .              31   Byte
  b50  . . . . . . . . . . . . .              32   Byte
  b51  . . . . . . . . . . . . .              33   Byte
  b52  . . . . . . . . . . . . .              34   Byte
  b53  . . . . . . . . . . . . .              35   Byte
  b54  . . . . . . . . . . . . .              36   Byte
  b55  . . . . . . . . . . . . .              37   Byte
  b56  . . . . . . . . . . . . .              38   Byte
  b57  . . . . . . . . . . . . .              39   Byte
  b58  . . . . . . . . . . . . .              3A   Byte
  b59  . . . . . . . . . . . . .              3B   Byte
  b60  . . . . . . . . . . . . .              3C   Byte
  b61  . . . . . . . . . . . . .              3D   Byte
  b62  . . . . . . . . . . . . .              3E   Byte
  b63  . . . . . . . . . . . . .              3F   Byte
__m512d  . . . . . . . . . . . .              40
  d0 . . . . . . . . . . . . . .               0   
  d1 . . . . . . . . . . . . . .               8   
  d2 . . . . . . . . . . . . . .              10   
  d3 . . . . . . . . . . . . . .              18   
  d4 . . . . . . . . . . . . . .              20   
  d5 . . . . . . . . . . . . . .              28   
  d6 . . . . . . . . . . . . . .              30   
  d7 . . . . . . . . . . . . . .              38   
__m512f  . . . . . . . . . . . .              40
  f0 . . . . . . . . . . . . . .               0   
  f1 . . . . . . . . . . . . . .               4   
  f2 . . . . . . . . . . . . . .               8   
  f3 . . . . . . . . . . . . . .               C   
  f4 . . . . . . . . . . . . . .              10   
  f5 . . . . . . . . . . . . . .              14   
  f6 . . . . . . . . . . . . . .              18   
  f7 . . . . . . . . . . . . . .              1C   
  f8 . . . . . . . . . . . . . .              20   
  f9 . . . . . . . . . . . . . .              24   
  f10  . . . . . . . . . . . . .              28   
  f11  . . . . . . . . . . . . .              2C   
  f12  . . . . . . . . . . . . .              30   
  f13  . . . . . . . . . . . . .              34   
  f14  . . . . . . . . . . . . .              38   
  f15  . . . . . . . . . . . . .              3C   
__m512i  . . . . . . . . . . . .              40
  i0 . . . . . . . . . . . . . .               0   DWord
  i1 . . . . . . . . . . . . . .               4   DWord
  i2 . . . . . . . . . . . . . .               8   DWord
  i3 . . . . . . . . . . . . . .               C   DWord
  i4 . . . . . . . . . . . . . .              10   DWord
  i5 . . . . . . . . . . . . . .              14   DWord
  i6 . . . . . . . . . . . . . .              18   DWord
  i7 . . . . . . . . . . . . . .              1C   DWord
  i8 . . . . . . . . . . . . . .              20   DWord
  i9 . . . . . . . . . . . . . .              24   DWord
  i10  . . . . . . . . . . . . .              28   DWord
  i11  . . . . . . . . . . . . .              2C   DWord
  i12  . . . . . . . . . . . . .              30   DWord
  i13  . . . . . . . . . . . . .              34   DWord
  i14  . . . . . . . . . . . . .              38   DWord
  i15  . . . . . . . . . . . . .              3C   DWord
__m512q  . . . . . . . . . . . .              40
  q0 . . . . . . . . . . . . . .               0   QWord
  q1 . . . . . . . . . . . . . .               8   QWord
  q2 . . . . . . . . . . . . . .              10   QWord
  q3 . . . . . . . . . . . . . .              18   QWord
  q4 . . . . . . . . . . . . . .              20   QWord
  q5 . . . . . . . . . . . . . .              28   QWord
  q6 . . . . . . . . . . . . . .              30   QWord
  q7 . . . . . . . . . . . . . .              38   QWord
__m512w  . . . . . . . . . . . .              40
  w0 . . . . . . . . . . . . . .               0   Word
  w1 . . . . . . . . . . . . . .               2   Word
  w2 . . . . . . . . . . . . . .               4   Word
  w3 . . . . . . . . . . . . . .               6   Word
  w4 . . . . . . . . . . . . . .               8   Word
  w5 . . . . . . . . . . . . . .               A   Word
  w6 . . . . . . . . . . . . . .               C   Word
  w7 . . . . . . . . . . . . . .               E   Word
  w8 . . . . . . . . . . . . . .              10   Word
  w9 . . . . . . . . . . . . . .              12   Word
  w10  . . . . . . . . . . . . .              14   Word
  w11  . . . . . . . . . . . . .              16   Word
  w12  . . . . . . . . . . . . .              18   Word
  w13  . . . . . . . . . . . . .              1A   Word
  w14  . . . . . . . . . . . . .              1C   Word
  w15  . . . . . . . . . . . . .              1E   Word
  w16  . . . . . . . . . . . . .              20   Word
  w17  . . . . . . . . . . . . .              22   Word
  w18  . . . . . . . . . . . . .              24   Word
  w19  . . . . . . . . . . . . .              26   Word
  w20  . . . . . . . . . . . . .              28   Word
  w21  . . . . . . . . . . . . .              2A   Word
  w22  . . . . . . . . . . . . .              2C   Word
  w23  . . . . . . . . . . . . .              2E   Word
  w24  . . . . . . . . . . . . .              30   Word
  w25  . . . . . . . . . . . . .              32   Word
  w26  . . . . . . . . . . . . .              34   Word
  w27  . . . . . . . . . . . . .              36   Word
  w28  . . . . . . . . . . . . .              38   Word
  w29  . . . . . . . . . . . . .              3A   Word
  w30  . . . . . . . . . . . . .              3C   Word
  w31  . . . . . . . . . . . . .              3E   Word


Types:

                N a m e                 Size    Attr

INNER2 . . . . . . . . . . . . .          16  INNER1


Segments and Groups:

                N a m e                 Size     Length   Align   Combine Class

FLAT . . . . . . . . . . . . . .        GROUP
_DATA  . . . . . . . . . . . . .        64 Bit   00000000 Para    Public  'DATA'
_TEXT  . . . . . . . . . . . . .        64 Bit   00000000 Para    Public  'CODE'


Symbols:

                N a m e                 Type       Value     Attr

@Arch  . . . . . . . . . . . . .        Number             0h 
@CodeSize  . . . . . . . . . . .        Number             0h 
@Cpu . . . . . . . . . . . . . .        Number           D5Fh 
@CurSeg  . . . . . . . . . . . .        Text   
@DataSize  . . . . . . . . . . .        Number             0h 
@Date  . . . . . . . . . . . . .        Text   03/21/22
@FileCur . . . . . . . . . . . .        Text   struct_typedef_align.asm
@FileName  . . . . . . . . . . .        Text   STRUCT_TYPEDEF_ALIGN
@Interface . . . . . . . . . . .        Number             0h 
@LastReturnType  . . . . . . . .        Number             0h 
@Model . . . . . . . . . . . . .        Number             7h 
@Platform  . . . . . . . . . . .        Number             1h 
@ProcLine  . . . . . . . . . . .        Number             0h 
@ProcName  . . . . . . . . . . .        Number             0h 
@ReservedStack . . . . . . . . .        Number             0h 
@Time  . . . . . . . . . . . . .        Text   18:43:34
@Version . . . . . . . . . . . .        Text   1000
@code  . . . . . . . . . . . . .        Text   _TEXT
@data  . . . . . . . . . . . . .        Text   FLAT
@stack . . . . . . . . . . . . .        Text   FLAT
__HJWASM__ . . . . . . . . . . .        Number            FFh 
__JWASM__  . . . . . . . . . . .        Number            D4h 
__UASM__ . . . . . . . . . . . .        Number            FFh 

struct_typedef_align.asm: 25 lines, 2 passes, 5 ms, 0 warnings, 0 errors
`
